flink-ml-lib/src/test/java/org/apache/flink/ml/feature/stringindexer/IndexToStringModelTest.java

```diff
@@ -0,0 +1,165 @@
+        env.setParallelism(4);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        tEnv = StreamTableEnvironment.create(env);
+
+        modelTable =
+                tEnv.fromDataStream(env.fromElements(new StringIndexerModelData(stringArrays)))
+                        .as("stringArrays");
+        predictTable =
+                tEnv.fromDataStream(env.fromCollection(Arrays.asList(Row.of(0, 3), Row.of(1, 2))))
+                        .as("inputCol1", "inputCol2");
+        predictTableWithUnseenValues =
+                tEnv.fromDataStream(
+                                env.fromCollection(
+                                        Arrays.asList(Row.of(0, 3), Row.of(1, 2), Row.of(4, 1))))
+                        .as("inputCol1", "inputCol2");
+    }
+
+    @Test
+    public void testPredictParam() {
+        IndexToStringModel indexToStringModel =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+        Table output = indexToStringModel.transform(predictTable)[0];
+        assertEquals(
+                Arrays.asList("inputCol1", "inputCol2", "outputCol1", "outputCol2"),
+                output.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testInputWithUnseenValues() {
+        IndexToStringModel indexToStringModel =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+        Table output = indexToStringModel.transform(predictTableWithUnseenValues)[0];
+        try {
+            IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+            fail();
+        } catch (Exception e) {
+            assertEquals(
+                    "The input contains unseen index: 4.",
+                    e.getCause().getCause().getCause().getCause().getCause().getMessage());
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testPredict() throws Exception {
+        IndexToStringModel indexToStringModel =
+                new IndexToStringModel()
+                        .setInputCols("inputCol1", "inputCol2")
+                        .setOutputCols("outputCol1", "outputCol2")
+                        .setModelData(modelTable);
+        Table output = indexToStringModel.transform(predictTable)[0];
+        List<Row> predictedResult =
+                IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        StringIndexerTest.verifyPredictionResult(expectedPrediction, predictedResult);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testSaveLoadAndPredict() throws Exception {
```

Comment:
nits: The code seems a bit too crowded as there is no line break in this method body.

Any chance we could make it less crowded? Maybe see `KMeansTest` or Spark's `KMeansSuite` for example?

Same for other tests.

Reply:
Thanks for pointing this out. I have done a pass for all code snippets in this PR.